contributors/devel/sig-architecture/import-boss.md

```diff
@@ -0,0 +1,69 @@
+# Using import-boss to enforce package import restrictions
+
+This document shows a way of enforcing package import restrictions using the `import-boss` tool.
+
+## Background
+
+The import-boss check runs against all pull requests submitted against the k/k repository. There are currently a good
+number of import restriction files, called `.import-restrictions` in the repository. The import restrictions files are
+specified in YAML format.
+
+## Rules
+
+Each import restrictions files contains rules. A Rule consists of three parts:
+
+1. A `SelectorRegexp`, to select the import paths that the rule applies to.
+2. A list of `AllowedPrefixes`.
+3. A list of `ForbiddenPrefixes`.
+
+All imports of a package are checked against each "rule" in the found restriction files, climbing up the directory tree
+until the import matches one of the rules. An import matches a rule of a matching selector if it matches at least one
+allowed prefix, but no forbidden prefix. If the import matches any of the rules, it is accepted.
+
+## Inverse Rules
+
+Reverse import rules help to define which packages in [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes)
+can import certain other packages, defined inside the latter. This allows to have fine-grained import restrictions for
```

Comment:
... In contrast to non-inverse rules, which are defined in importing packages, these inverse rules are defined in imported packages.